src/Button/Button.spec.tsx

```diff
@@ -64,6 +64,29 @@ test("when disabled, button does not call click handlers", () => {
   expect(rootElementOnClick).not.toHaveBeenCalled();
 });
 
+test("when disabled, button with 'as={}' does not call click handlers", () => {
```

Comment:
This needs a better description. Disabling a button when using `as={<button>}` works correctly; this needs to describe a more specific case in which it doesn't work correctly.